services/economics-rewards/src/delegators.ts

```diff
@@ -0,0 +1,87 @@
+import { DelegationData, Delegator } from './types';
+import { getContractInstance } from './contract';
+import { saveDelegator } from './database';
+
+const getDelegations = async (
+  delegator: string,
+  blockNumber: number,
+  index: number
+) => {
+  const { dposContract, provider } = await getContractInstance();
+  const { hash, timestamp } = await provider.getBlock(blockNumber);
+
+  const res: {
+    delegations: DelegationData[];
+    end: boolean;
+  } = await dposContract.getDelegations(delegator, index, {
+    blockTag: blockNumber,
+  });
+  const formattedDelegations: Delegator[] = await Promise.all(
+    res.delegations.map(async (delegationData: DelegationData) => {
+      const formatted = {
+        blockNumber: blockNumber,
+        blockTimestamp: timestamp,
+        blockHash: hash,
+        delegator: delegator,
+        validator: delegationData.account,
+        stake: delegationData?.delegation?.stake?.toString(),
+        rewards: delegationData?.delegation?.rewards?.toString(),
+      };
+      await saveDelegator(formatted);
```

Comment:
```suggestion
      const savedDelegator = await saveDelegator(formatted);
      return savedDelegator as Delegator
```
Would be good to return the persisted value.